### PR TITLE
Feature/api fetch

### LINKED ISSUE #16: Explore page
As a user,
I want a explore page,
so I can find new coins.

- [x] Should show the marquee component
- [x] Should show the list element
- [x] Should show category-coins
- [x] Should show a max. of 10 coins
- [x] Should have a "load more" action

### LINKED ISSUE #18: Implement API fetch
As a user,
I want current coin-data,
so I can inform myself.

- [ ] Should be able to display market-data based on the coin-category
- [ ] Should use axios for fetching
- [ ] Should use zustand for state-handling

### PR DESCRIPTION
**Pull request by konrascher3**
## Motivation
User should be able to change coin-categories. The data-grid should then display the coins based on the selected category. Due to mui-limitations, only up to 100 coins per data-grid can be displayed at once.

* Added API-fetch-routes
* Tabs change the fetch-route dynamically
* Zustand is used for state-handling
* Added "load more"-button up to 100 coins

## Issues closed
* closes #18 
* closes #16

### CODE REVIEW

File: src/ions/hooks/state/useStore.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import create from "zustand";
+import axios from "axios";
+
+const useStore = create((set) => ({
+	data: null,
+	loading: false,
+	error: null,
+	coins: null,
+	endpoint: "",
+	url:"",
```
Comment (konrascher3) [inline src/ions/hooks/state/useStore.js:10]: Remove unused keys

File: src/molecules/tab-bar/TabBar.jsx [hunk 2]
```diff
@@ -41,10 +42,53 @@ function a11yProps(index) {
...
+
 	const [value, setValue] = React.useState(0);
+	const fetchData = useStore((state) => state.fetchData)
+	const loading = useStore((state)  => state.loading)
 
 	const handleChange = (event, newValue) => {
 		setValue(newValue);
+		switch (newValue) {
+			case 0:
+				console.log("top-coins fetching")
```
Comment (konrascher3) [inline src/molecules/tab-bar/TabBar.jsx:64]: Remove console.logs

File: src/pages/index.jsx [hunk 3]
```diff
@@ -94,9 +95,22 @@ const columns = [
...
+
+	// Reset page-size if category switches
+	useEffect(()=>{setPageSize(initialPageSize)}, [data])
 
+	// Initial fetch if data = null
+	if(data === null) {
+		fetchData("https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=100&page=1&sparkline=false")
+	}
+	console.log(data?.length);
+	console.log(pageSize)
```
Comment (konrascher3) [inline src/pages/index.jsx:113]: Remove console.logs

File: src/ions/hooks/state/useStore.js (hunk 1 above)
Comment (marcbendixen) [reply src/ions/hooks/state/useStore.js]: Good point 🙂

File: src/molecules/tab-bar/TabBar.jsx [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import Box from "@mui/material/Box";
 import React from "react";
+import useStore from "../../ions/hooks/state/useStore";
 
 // MUI Imports
 import PropTypes from 'prop-types';
```
Comment (marcbendixen) [inline src/molecules/tab-bar/TabBar.jsx:6]: Nice! I like that you use `prop-types` 👍

**Review by marcbendixen — APPROVED** [review]
Well done 🎉 
lgtm

**Pull request merged**